### DISCUSSION TITLE
Final classes

### DISCUSSION THREAD

**Discussion by Obi-wan-koban** [tdlight-team/tdlight-java#95] [Q&A]
Hi, I already made a working application using the official example for the java version of tdlib, and was now redesigning it, incorporating the tdlight library instead of the manually compiled java source files. 
There is however one very big frustration that I constantly need to keep working around: Final classes.
Why has there been opted to always use final classes? 
I don't really see advantages to this for people who want to use your library?

**Comment by cavallium** [comment]
A lot of people kept breaking the library thinking that they needed to extend some internal classes that should not be touched, so I made them private.
What final classes do you think that I should make modifiable again? Are you talking about the client classes, or the TdApi classes?

**Reply by cavallium** [reply]
Also, please note that the "simple client" is just a very simple client for beginners. If you really need to make a more advanced or specialized client, you should directly use the client manager: [AdvancedExample.java](https://github.com/tdlight-team/tdlight-java/blob/master/example/src/main/java/it/tdlight/example/AdvancedExample.java)
It gives you direct access to the TDLib APIs without interfering with your calls, but it's obviously more difficult to use.

**Reply by Obi-wan-koban** [reply]
In my first version, I extended the TdApi.Chat object and persisted that object into the database. Disabling the default database that is built in tdlib and only using my own database. It was rather useful, since I could directly get the appropriate entity (if it existed).

**Reply by cavallium** [reply]
Why can't you wrap it? The classes inside TdApi.java should not be extended
